crates/core/src/types/set_prices_params.rs

```diff
@@ -0,0 +1,130 @@
+use serde::Serialize;
+use starknet::core::types::{EthAddress, FieldElement};
+
+use crate::types::field_element::IntoFieldElementVec;
+
+#[derive(Debug, Serialize)]
+pub struct SetPricesParams {
+    pub signer_info: u128,
+    pub tokens: Vec<EthAddress>,
+    pub compacted_min_oracle_block_numbers: Vec<u64>,
+    pub compacted_max_oracle_block_numbers: Vec<u64>,
+    pub compacted_oracle_timestamps: Vec<u64>,
+    pub compacted_decimals: Vec<u128>,
+    pub compacted_min_prices: Vec<u128>,
+    pub compacted_min_prices_indexes: Vec<u128>,
+    pub compacted_max_prices: Vec<u128>,
+    pub compacted_max_prices_indexes: Vec<u128>,
+    pub signatures: Vec<FieldElement>,
+    pub price_feed_tokens: Vec<EthAddress>,
+}
+
+impl From<&SetPricesParams> for Vec<FieldElement> {
+    fn from(item: &SetPricesParams) -> Self {
+        let mut field_elements = Vec::new();
+
+        field_elements.push(FieldElement::from(item.signer_info));
+        field_elements.extend(item.tokens.as_field_element_vec());
+        field_elements.extend(
+            item.compacted_min_oracle_block_numbers
+                .as_field_element_vec(),
+        );
+        field_elements.extend(
+            item.compacted_max_oracle_block_numbers
+                .as_field_element_vec(),
+        );
+        field_elements.extend(item.compacted_oracle_timestamps.as_field_element_vec());
+        field_elements.extend(item.compacted_decimals.as_field_element_vec());
+        field_elements.extend(item.compacted_min_prices.as_field_element_vec());
+        field_elements.extend(item.compacted_min_prices_indexes.as_field_element_vec());
+        field_elements.extend(item.compacted_max_prices.as_field_element_vec());
+        field_elements.extend(item.compacted_max_prices_indexes.as_field_element_vec());
+        field_elements.extend(item.signatures.clone());
+        field_elements.extend(item.price_feed_tokens.as_field_element_vec());
+
+        field_elements
+    }
+}
```

Comment:
Ah right, nice catch! Totally forgot.

I introduced a new function for `Vec<FieldElement>` extend_with_len that adds the size.

I don't think it's necessary to add the size for unique elements too since we should be able to tell where they are but I can add that too if you think it's necessary (for `signer_info` I didn't add the len `1` before the element).

Reply:
Thanks, i've added some comments.
I think after that it'll be fine for this PR

And yes, no need to add the len for elements which were not a vec.